src/gretel_trainer/trainer.py

```diff
@@ -0,0 +1,156 @@
+"""Main Trainer Module"""
+
+from collections import namedtuple
+from enum import Enum
+import json
+import logging
+import pandas as pd
+import pkgutil
+import yaml
+
+from gretel_client import configure_session, ClientConfig
+from gretel_client.projects import create_or_get_unique_project
+from gretel_client.projects.models import read_model_config
+from gretel_client.projects.jobs import Status
+from gretel_synthetics.utils.header_clusters import cluster
+
+from gretel_trainer import strategy, runner
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel("DEBUG")
+
+
+class ExtendedEnum(Enum):
+    """Utility class for Model enum"""
+
+    @classmethod
+    def get_types(cls):
+        return list(map(lambda c: c.name, cls))
+
+    @classmethod
+    def get_config(cls, model: str):
+        return cls[model].config
+
+
+class Model(namedtuple("Model", "config"), ExtendedEnum):
+    """Enum to pair valid models and configurations"""
+
+    GretelLSTM = "templates/gretel_lstm.yaml"
+    GretelCTGAN = "templates/gretel_ctgan.yaml"
+
+
+class Trainer:
+    """Automated generative training and sampling tool
+
+    Args:
+        project_name (str, optional): Gretel project name. Defaults to "trainer".
+        max_header_clusters (int, optional): Max number of clusters per batch. Defaults to 20.
+        max_rows (int, optional): Max number of rows per batch. Defaults to 50000.
+        model_type (str, optional): Options include ["GretelLSTM", "GretelCTGAN"]. Defaults to "GretelLSTM".
+        model_kwargs (dict, optional): Modify model configuration settings by key. E.g. {'epochs': 20}
+    """
+
+    def __init__(
+        self,
+        project_name: str = "trainer",
+        max_header_clusters: int = 20,
+        max_rows: int = 50000,
+        model_type: str = "GretelLSTM",
+        model_kwargs: dict = {}
+    ):
+
+        configure_session(api_key="prompt", cache="yes", validate=True)
```

Comment:
Issue: if I already have a client configured with this force me to re-authenticate? We may want to first check if the client is already configured and able to make api calls first before forcing a re-auth

Reply:
If you already have a client configured it will use your existing credentials and will **not** prompt you to re-auth.